---
title: "4.- TCGA_BRCA_CNV"
author: "Nuria Moragas"
date: "July 2022"
output:
  html_document:

---


**WEB BASE:**

- http://bioconductor.org/packages/release/workflows/vignettes/TCGAWorkflow/inst/doc/TCGAWorkflow.html#Genomic_analysis
- https://openaccess.uoc.edu/bitstream/10609/126978/6/mlozanogarciaTFM0121memoria.pdf
- https://stdworkflow.com/277/tcga-data-analysis-combat-cnv-and-mutation
-https://www.biostars.org/p/311199/ (Kevin Blighe)
- https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5302158/pdf/f1000research-5-11063.pdf
(GITHUB:https://github.com/nanxstats/dockflow/blob/master/workflow/tcga/src/vignettes/index.Rmd )

**INDEX:**  

# 4.1 - Work environment preparation

This part include:
  - 4.1.1 - Directory Definition
  - 4.1.2 - Library Call
  - 4.1.3 - Color Palette
  
# 4.2 - CNV Data Download
# 4.3 - Pre-procesing
# 4.4 - EXPLORATORY ANALYSIS
# 4.5 - CNV identification
# 4.6 - Gene annotation of recurrent CNV


-------------------------------------------------------------

# 4.1 -Work environment preparation
## 4.1.1 - Directory Definition


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, comment = NULL, cache = TRUE)
```

```{r message=FALSE, warning=FALSE}

#___________________Directori definition____________________#

setwd("C:/Users/nmora/Desktop/TFM_R Scrips/Scrips_R")

```

## 4.1.2 - Library Call

```{r message=FALSE, warning=FALSE}
#_________________________Packages_______________________#

#Load packages
#TCGA
library(TCGAbiolinks)
library(dplyr)
library(ggpubr) #statistics
library(knitr)

library(SummarizedExperiment)
library(RColorBrewer)
library(ComplexHeatmap)

library(gaia)
library(stringr)
library(GenomicRanges)
library(base)

```

## 4.1.3 - Color Palette

```{r}
 #...................................COLORS........................

colors_Gr_Bw <- c( "#35978F", "#80CDC1", "#C7EAE5", "#FFFFFF" , "#F6E8C3", "#DFC27D", "#BF812D" ,"#8C510A")

```

                # --------------------------  #
 
 
 
# 4.2 - CNV Data Download

```{r}

#1.- Find Primary Tumour and Metastasi CNV samples - IDC
barcode_TM_IDC = readRDS(file ="0.-barcode_TM_IDC.rda")
barcode_TP_IDC = readRDS(file ="0.-barcode_TP_IDC.rda")

## Primary Tumor - TP

CNV_TP <-  GDCquery(
    project = "TCGA-BRCA", 
    barcode = barcode_TP_IDC, #Dif IDC samples
    data.category = "Copy Number Variation",
    data.type = "Copy Number Segment", 
    legacy = T,
    sample.type = c("Primary Tumor"))

#To visualize the query results in a more readable way, we can use the command getResults.
            brca_CNV_TP = getResults(CNV_TP) # make results as table
            # head(lihc_res) # data of the first 6 patients.
          
            dim((brca_CNV_TP)) #[1] 781   29
            brca_CNV_TP
            summary(factor(brca_CNV_TP$sample_type)) 
            

#see the barcodes with
TP_barcode <- CNV_TP$results[[1]]$cases
TP_barcode            
            
# Download CNV data.
GDCdownload(CNV_TP, method = "api")


# Finally, lets load the CNV results as values
BRCA_CNV_TP <- GDCprepare(CNV_TP)


#Look inside the data
table(BRCA_CNV_TP$Sample)

dim(BRCA_CNV_TP) # [1] 611725    7

BRCA_CNV_TP
            
#save
saveRDS(BRCA_CNV_TP, file = "4.-BRCA_CNV_TP.RDS", compress = FALSE)

#load:
BRCA_CNV_TP = readRDS(file = "4.-BRCA_CNV_TP.RDS")    
            

summary(BRCA_CNV_TP$Segment_Mean)
```

```{r}
## Metastasis - TM
CNV_TM <-  GDCquery(
    project = "TCGA-BRCA", 
    barcode = barcode_TM_IDC,
    data.category = "Copy Number Variation",
    data.type = "Copy Number Segment", 
    legacy = T,
    sample.type = c("Metastatic"))


#To visualize the query results in a more readable way, we can use the command getResults.
            brca_CNV_TM = getResults(CNV_TM) # make results as table
            # head(lihc_res) # data of the first 6 patients.
          
            dim((brca_CNV_TM)) #[1]6 27 
            brca_CNV_TM$cases
            summary(factor(brca_CNV$sample_type)) #Metastatic Primary Tumor         
#see the barcodes with
TM_barcode <- CNV_TM$results[[1]]$cases
            
            
# Download CNV data.
GDCdownload(CNV_TM, method = "api")


# Finally, lets load the CNV results as values
BRCA_CNV_TM <- GDCprepare(CNV_TM)


#Look inside the data
table(BRCA_CNV_TM$Sample)

dim(BRCA_CNV_TM) # [1] 4296    7
table(BRCA_CNV_TM$Sample)

 #save
saveRDS(BRCA_CNV_TM, file = "4.-BRCA_CNV_TM.RDS", compress = FALSE)

#load:
BRCA_CNV_TM = readRDS(file = "4.-BRCA_CNV_TM.RDS")  


summary(BRCA_CNV_TM$Segment_Mean)
```


```{r}
#Add column with TP or TM - Primary Sample o Metastatic
#"Metastatic" "Primary Tumor
#sample_type

BRCA_CNV_TM$sample_type <- c("Metastatic")

BRCA_CNV_TP$sample_type <- c("Primary Sample")


#shortLetterCode - Metastatic(TM) Primary Tumor(TP) 

BRCA_CNV_TM$shortLetterCode <- c("TM")
BRCA_CNV_TP$shortLetterCode <- c("TP")

```

```{r}
#Add the two df: Tp i TM#

BRCA_CNV_TM_TP_IDC <- rbind(BRCA_CNV_TM, BRCA_CNV_TP)
table(BRCA_CNV_TM_TP_IDC$sample_type)
BRCA_CNV_TM_TP_IDC
```



```{r}
#save
saveRDS(BRCA_CNV_TM_TP_IDC, file = "4.-BRCA_CNV_TM_TP_IDC.RDS", compress = FALSE)

#save as a txt
write.table(BRCA_CNV_TM_TP_IDC,"4.BRCA_CNV_TM_TP_IDC.txt",sep="\t",row.names=FALSE)

#load:
BRCA_CNV_TM_TP_IDC = readRDS(file = "4.-BRCA_CNV_TM_TP_IDC.RDS")

```

                # --------------------------  #
    

# 4.3 - Pre-procesing

```{r}

## 3.Filtrado de Missing values:
## A.- remove probes with NA
probe.na <- rowSums(is.na(BRCA_CNV_TM_TP_IDC ))

table(probe.na == 0)


## 4. Control de Calidad
#These data have already been normalized and noted (level 3) previos normalization.

df <- data.frame(BRCA_CNV_TM_TP_IDC, BRCA_CNV_TM_TP_IDC$Sample)
```

```{r}
#OPCIO 1 - Density Plot

png(file="Plots_results/4.0.- Density_plot_CNV.png" #,  width =  500 , height = 400 , units = "px" 
    )
ggplot(BRCA_CNV_TM_TP_IDC, aes(x= BRCA_CNV_TM_TP_IDC$Segment_Mean, colour = BRCA_CNV_TM_TP_IDC$sample_type ))+
  geom_density()+
  
  #Change the order of items in the legend
  scale_x_discrete(limits=c("Primary Sample", "Metastatic"))+
  
  #Change colour:
  scale_color_manual(values=c("#80CDC1", "#DFC27D"))+ 
  
  theme_classic()

dev.off()

```


## 4.4 - EXPLORATORY ANALYSIS

```{r}
#Statistics
compare_means(Segment_Mean ~ sample_type, data = BRCA_CNV_TM_TP_IDC)


group_by(BRCA_CNV_TM_TP_IDC, sample_type) %>%
  summarise(
    count = n(),
    mean = mean(Segment_Mean, na.rm = TRUE),
    sd = sd(Segment_Mean, na.rm = TRUE)
  )

t.test(Segment_Mean ~ sample_type, data = BRCA_CNV_TM_TP_IDC, var.equal = TRUE)

```

```{r}
my_comparisons <- list( c("Primary Sample", "Metastatic") )


png(file="Plots_results/4.1.- CNV_count_mean_comparison.png" #,  width =  500 , height = 400 , units = "px" 
    )

ggplot(BRCA_CNV_TM_TP_IDC, aes(x = sample_type, y = Segment_Mean)) +            # Applying ggplot function
  geom_violin(aes(fill =  sample_type)) +
#Add boxplot  
 geom_boxplot(width=0.2)+
  
  labs(y = "Segment Mean", x = "Sample Type")+
  
  geom_hline(yintercept= 0.3, linetype="dashed", color = "azure3") +
  geom_hline(yintercept= -0.3, linetype="dashed", color = "azure3") +
  
 
 
#Change the order of items in the legend
  scale_x_discrete(limits=c("Primary Sample", "Metastatic"))+
  
#Change colour:
scale_color_manual(values=c("#80CDC1", "#DFC27D"))+ #COntron
scale_fill_manual(values=c("#80CDC1", "#DFC27D"))+  
  theme_classic()+
  
#Statistics  
  stat_compare_means(comparisons = my_comparisons, method = "wilcox.test")
   
dev.off()
```



```{r}
# Nº de events per CNV gain /loss per sample - PRIMARY SAMPLE

#o aquest junt amb l'altre de Metastasis

#Primary Tumor
BRCA_CNV_TP <- BRCA_CNV_TM_TP_IDC %>% filter(sample_type == "Primary Sample")
BRCA_CNV_TP <- BRCA_CNV_TP [!(BRCA_CNV_TP$Chromosome %in% c("Y")),]
BRCA_CNV_TP



BRCA_CNV_TP$color <- ifelse(BRCA_CNV_TP$Segment_Mean<0, 'loss','gain')


png(file="Plots_results/4.2.- CNV_PT_gain_loss.png" #,  width =  500 , height = 400 , units = "px" 
    )

ggplot(BRCA_CNV_TP, aes(x=Chromosome, y=Segment_Mean)) +
       geom_bar(stat="summary",
                width=0.8, aes(fill=color)) +
  #Title
  labs(title="Plot of Primary Sample - CNV gain/loss", 
         x="Chromosome", y = "Segmeant Mean")+
  #Change the order of items in the legend
  scale_x_discrete(limits=c("1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16", "17", "18", "19", "20", "21", "22", "X"))+
       scale_fill_manual(values=c(gain="#80CDC1",loss="#DFC27D")) +
       geom_line(aes(y=0), color='black') + theme_classic()

dev.off()

```


```{r}
# Nº de events per CNV gain /loss per sample - Metastatic

BRCA_CNV_TM <- BRCA_CNV_TM_TP_IDC %>% filter(sample_type == "Metastatic")
BRCA_CNV_TM <- BRCA_CNV_TM [!(BRCA_CNV_TM$Chromosome %in% c("Y")),]
BRCA_CNV_TM$color <- ifelse(BRCA_CNV_TM$Segment_Mean<0, 'loss','gain')
BRCA_CNV_TM


png(file="Plots_results/4.2.- CNV_MT_gain_loss.png" #,  width =  500 , height = 400 , units = "px" 
    )

ggplot(BRCA_CNV_TM, aes(x=Chromosome, y=Segment_Mean)) +
       geom_bar(stat="summary",
                width=0.8, aes(fill=color)) +
   #Title
  labs(title="Plot of Metastatic - CNV gain/loss", 
         x="Chromosome", y = "Segmeant Mean")+
  #Change the order of items in the legend
  scale_x_discrete(limits=c("1", "2", "3", "4", "5", "6", "7", "8", "9","10", "11", "12", "13", "14", "15", "16", "17", "18", "19", "20", "21", "22", "X"))+
       scale_fill_manual(values=c(gain="#80CDC1",loss="#DFC27D")) +
       geom_line(aes(y=0), color='black') + theme_classic()
dev.off()

```

```{r}

# Change the colors manually
ggplot(data = BRCA_CNV_TM_TP_IDC, aes(x=Chromosome, y=Segment_Mean, fill=sample_type)) +
geom_bar(stat="identity", color="black", position=position_dodge())+
  theme_minimal()+
  scale_fill_manual(values=c("#80CDC1","#DFC27D"))

```


# 4.5 - CNV identification

2 part, a common one for both TP and TM, and a second specific for each type of sample.

  * A.- Common part
  * B.- Metastatic 

## A.- COMMON PART
```{r}
# --------------------- A.Common part -----------------


#1.- Library

#2.- Load marker matrix
#https://gdc.cancer.gov/about-data/gdc-data-processing/gdc-reference-files
#The SNP6 GRCh38 Liftover Probeset File for Copy Number Variation Analysis was download from the GDC website - SNP6.na35.liftov
#SNP6 GRCh38 Liftover Probeset File for Copy Number Variation Analysis
#snp6.na35.liftoverhg38.txt.zip

markersMatrix <- read.table("4.-snp6.na35.remap.hg38.subset.txt", sep = "\t", head = T)
markersMatrix

#3.- Process/Format Markers matrix
##Re-name colnames
colnames(markersMatrix)[1:3] = c("Probe.Name", "Chromosome", "Start")


##Substitute CHr X and Y with 23 and 24. And Convert Chromosome col to integr
markersMatrix$Chromosome = as.character(markersMatrix$Chromosome)
markersMatrix[markersMatrix$Chromosome == "X", "Chromosome"] = 23
markersMatrix[markersMatrix$Chromosome == "Y", "Chromosome"] = 24
markersMatrix$Chromosome = as.integer(markersMatrix$Chromosome )

##There are duplicated markers?
markerID <- paste(markersMatrix$Chromosome, markersMatrix$Start, sep =":")
print(table(duplicated(markerID) ))
      #FALSE    TRUE 
      #1837320     187
print(table(duplicated(markersMatrix$Probe.Name)))
#FALSE 
#1837507
#187 duplicated makers with different names. 

#Keep no duplicated markers
markersMatrix <- markersMatrix[!duplicated(markerID),] 
markerID <- paste(markersMatrix$Chromosome, markersMatrix$Start, sep =":")
print(table(duplicated(markerID) ))
#FALSE 
#1837320

# Create the markers object
markers_obj <- load_markers(markersMatrix)


#save
saveRDS(markers_obj, file = "4.markers_obj.RDS", compress = FALSE)



#load:
markers_obj = readRDS(file = "4.markers_obj.RDS")


#----------

```

## B.- CNV METASTATIC SAMPLES

  * Part I: load GAIA
  * Part II: Run GAIA
  * Part III: Gene annotation of recurrent CNV
  
### Part I: load GAIA


```{r}

#-------------------------- B.- METASTATIC SAMPLES -----------------------

##___________________________________ PART I _________________________________________


#Part I - download segmented sCNA data for any TCGA cohort from Broad Institute's FireBrowse server and identify recurrent sCNA regions in these with GAI


#2.- Load data
BRCA_CNV_TM_TP_IDC = readRDS(file = "4.-BRCA_CNV_TM_TP_IDC.RDS")


BRCA_CNV_TM_gaia <- BRCA_CNV_TM_TP_IDC %>% filter(sample_type == "Metastatic")
BRCA_CNV_TM_gaia <- BRCA_CNV_TM_gaia[,-1]
BRCA_CNV_TM_gaia <- BRCA_CNV_TM_gaia[,-8]
BRCA_CNV_TM_gaia <- BRCA_CNV_TM_gaia[,c(6,1,2,3,4,5,7)]
BRCA_CNV_TM_gaia <- BRCA_CNV_TM_gaia[,-7]



#_________________________ A.- CNV MATRIX________________________
#4.- Process/Format CNV matrix

cnvMatrix_TM = cbind(BRCA_CNV_TM_gaia, Label = NA)
##Es crea una nova variant "Label" on 0 son deleccions i 1 amplificacions
cnvMatrix_TM[cnvMatrix_TM[, "Segment_Mean"]< -0.3, "Label"] = 0
cnvMatrix_TM[cnvMatrix_TM[, "Segment_Mean"]> 0.3, "Label"] = 1
cnvMatrix_TM <- cnvMatrix_TM[!is.na(cnvMatrix_TM$Label),]

##Segment "Mean Variable" delation and change colnames
cnvMatrix_TM <- cnvMatrix_TM[,-6]
colnames(cnvMatrix_TM) <- c("Sample.Name", "Chromosome", "Start", "End", "Num.of.Markers", "Aberration")


##Substitute CHr X and Y with 23 and 24. And convert chrm colum from caracter to 
cnvMatrix_TM[cnvMatrix_TM$Chromosome == "X", "Chromosome"] = 23
cnvMatrix_TM[cnvMatrix_TM$Chromosome == "Y", "Chromosome"] = 24
cnvMatrix_TM$Chromosome = as.integer(cnvMatrix_TM$Chromosome)



cnvMatrix$Num.of.Markers <- cnvMatrix$End-cnvMatrix$Start+1


#_________________________ B.- GAIA ________________________

#6.- Load data into R object for Gaia


n_samples_TM <-length(unique(cnvMatrix_TM$Sample.Name))
n_samples_TM
#----------

#save

saveRDS(n_samples_TM, file = "4.n_samples_TM.RDS", compress = FALSE)
saveRDS(cnvMatrix_TM, file = "4.cnvMatrix_TM.RDS", compress = FALSE)

#load:
markers_obj = readRDS(file = "4.markers_obj.RDS")
n_samples_TM = readRDS(file = "4.n_samples_TM.RDS")
cnvMatrix_TM = readRDS(file = "4.cnvMatrix_TM.RDS")
#----------

cnv_obj_TM <- load_cnv(cnvMatrix_TM, markers_obj, n_samples_TM)


saveRDS(cnv_obj_TM, file = "4.cnv_obj_TM.RDS", compress = FALSE)
#cnv_obj_TM = readRDS(file = "4.cnv_obj_TM.RDS")

```

### Part II: Run GAIA

```{r}
##___________________________________ PART II _________________________________________

#Part II - plot recurrent sCNA gains and losses from GAIA

##----------------- 7.- Run Gaia ----------

set.seed(400)

suppressWarnings({
  TM_gaia_results <- runGAIA(cnv_obj_TM, markers_obj, output_file_name = "4.-CNV_BRCA_TM_GAIA_results.txt", 
               
      # Specify the type of variation to be analyzed, -1 to analyze all the variations             
      aberrations = -1, 
      ## Specify the chromosomes to be analyzed, the default is -1, which means all chromosomes
      chromosomes = -1, 
      ## Using approximate methods can speed up the calculation
      #approximation = TRUE, 
      # Set the number of iterations
      num_iterations = 10)
      #thereshold =  )# - default=0.25
  })


      
##----------------- 8.- tidy output ----------
TM_gaia_results  <- read.table("4.-CNV_BRCA_TM_GAIA_results.txt", header =T, sep = "\t" )
class(TM_gaia_results)

TM_gaia_results


TM_gaia_results[TM_gaia_results$Type == "Del", "Type"] = 0
TM_gaia_results[TM_gaia_results$Type == "Amp", "Type"] = 1
TM_gaia_results$Type = as.integer(TM_gaia_results$Type)


threshold <- 0.25

## Convertir nom columnes en numero | #Convert the recurrent aberrations to numeric
RecCNV <- t(apply(TM_gaia_results,1,as.numeric))
colnames(RecCNV) <- colnames(TM_gaia_results)

#Add "score" column
RecCNV <- cbind(RecCNV, score = 0)

##Determine the minimum Q value that's not equal to 0
minval <- format(min(RecCNV[RecCNV[,"q.value"] != 0,"q.value"]), scientific = FALSE)
minval <- substring(minval,1, nchar(minval) - 1)

##Replace Q values of 0 with the minimum, non-zero value
RecCNV[RecCNV[,"q.value"] == 0,"q.value"] <- as.numeric(minval)

###Set the score to equal -log base 10 of the Q value
RecCNV[,"score"] <- sapply(RecCNV[,"q.value"],function(x) -log10(as.numeric(x)))


RecCNV[RecCNV[,"q.value"] == as.numeric(minval),]



##----------------- 9.- PLOT ----------

##RecCNV no ordenada com necessita la funcio gaiaCNVplot. Eliminar i re ordenar
RecCNV_2 <- as.data.frame(RecCNV)
RecCNV_2 <- RecCNV_2[,-8]
RecCNV_2 <- RecCNV_2[,-7]
RecCNV_2 <- RecCNV_2[,-6]
RecCNV_2 <- RecCNV_2[,c(2,1,3,4,6,5)]


RecCNV_2$Start = as.integer(RecCNV_2$Start)
RecCNV_2$End = as.integer(RecCNV_2$End)


RecCNV_2<-RecCNV_2[!(RecCNV_2$Chromossome =="24"),]
table(RecCNV_2$Chromossome)



##Renames:
colnames(RecCNV_2) <- c("Chromossome", "Aberration Kind",  "Region Start", "Region End", "score", "q-value" )


#Save
save(TM_gaia_results, RecCNV_2, threshold, file = "4.-BRCA_CNV_TM_results.rda") 
load(file = "4.-BRCA_CNV_TM_results.rda")
```

```{r}


#Create a function for plotting the recurrent copy number variants

gaiaCNVplot_2 <- function (calls, threshold=0.3, main="") {
    Calls <- calls[order(calls[,"Region Start"]),]
    Calls <- Calls[order(Calls[,"Chromossome"]),]
    rownames(Calls) <- NULL
    Chromo <- Calls[,"Chromossome"]
    Gains <- apply(Calls,1,function(x) ifelse(x["Aberration Kind"]==1, x["score"], 0))
    Losses <- apply(Calls, 1,function(x) ifelse(x["Aberration Kind"]==0, x["score"], 0))
    plot(Gains, ylim=c(-max(Calls [,"score"]+2), max(Calls[,"score"]+2)), type="h", col="#80CDC1", lwd= 3:5,
         
         xlab="Chromosome", ylab=expression("-log"[10]~italic(Q)~"value"), main=main, cex.main=4, xaxt="n", font=2, font.axis=2, font.lab=2, font.axis=2, cex.axis= 1.3, cex.lab = 1.5) 
    
    #cex.axis = mida lletra eix y, cex.lab = mida lletra nom dels eixos,
    
    points(-(Losses), type="h", col="#DFC27D", lwd= 3:5)
    abline(h= 0, cex=4)
    abline(h=-log10(threshold), col="black", cex=4, main="test", lty=6, lwd=2)
    abline(h=log10(threshold), col="black", cex=4, main="test", lty=6, lwd=2)
    uni.chr <- unique(Chromo)
    temp <- rep(0, length(uni.chr))

    for (i in 1:length(uni.chr)) {
        temp[i] <- max(which(uni.chr[i] == Chromo))
    }

    for (i in 1:length(temp)) {
        abline(v = temp[i], col = "black", lty = "dashed", )
    }

    nChroms <- length(uni.chr)

    begin <- c()

    for (d in 1:nChroms) {
        chrom <- sum(Chromo == uni.chr[d])
        begin <- append(begin, chrom)
    }

    temp2 <- rep(0, nChroms)

    for (i in 1:nChroms) {
        if (i == 1) {
            temp2[1] <- (begin[1] * 0.5)
        }
        else if (i > 1) {
            temp2[i] <- temp[i - 1] + (begin[i] * 0.5)
        }
    }

    uni.chr[uni.chr==23] <- "X"
    uni.chr[uni.chr==24] <- "Y"

    for (i in 1:length(temp)) {
        axis(1, at = temp2[i], labels = uni.chr[i], cex.axis = 1)
    }

    legend("topright", y.intersp=0.8, c("Amplification"), pch=15, col=c("#80CDC1"), text.font=2)
    legend("bottomright", y.intersp=0.8, c("Deletion"), pch=15, col=c("#DFC27D"), text.font=2)
}


png(file="Plots_results/4.3.- CNV_TM_gaia.png" #,  width =  500 , height = 400 , units = "px" 
    )
gaiaCNVplot_2(RecCNV_2,threshold)
dev.off()


table(RecCNV_2$`Aberration Kind`)
#DEl  Amp
#  0   1 
#256 195 

dim(RecCNV_2)

table(RecCNV_2$Chromossome)
#1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 
#12 12 18 12 26 26 18 14 20 23 14 26 14 28 32 30 16 12 27 20 20 17 14
```



### Part III: Gene annotation of recurrent CNV

```{r , eval = TRUE, message=FALSE,warning=FALSE, include=FALSE}
library(GenomicRanges)
```

```{r , eval = FALSE, message=FALSE,warning=FALSE, include=TRUE}


##___________________________________ PART III _________________________________________


# Get gene information from GENCODE using biomart
genes <- TCGAbiolinks:::get.GRCh.bioMart(genome = "hg19") 
genes <- genes[genes$external_gene_name != "" & genes$chromosome_name %in% c(1:22,"X","Y"),]
             
genes[genes$chromosome_name == "X", "chromosome_name"] <- 23
genes[genes$chromosome_name == "Y", "chromosome_name"] <- 24
genes$chromosome_name = as.integer(genes$chromosome_name)


genes <- genes[order(genes$start_position),]
genes <- genes[order(genes$chromosome_name),]

#Reordenar
genes <- genes[,c("external_gene_name", "chromosome_name", "start_position","end_position")]
genes
#Renombrar
colnames(genes) <- c("GeneSymbol","Chr","Start","End")

#Make GRanges object
genes_GR <- makeGRangesFromDataFrame(genes,keep.extra.columns = TRUE)
save(genes_GR,genes,file = "4.- BRCA_CNV_TM_genes_GR.rda", compress = "xz")

load(file = "4.- BRCA_CNV_TM_genes_GR.rda")

```

```{r , echo=TRUE, message=FALSE,warning=FALSE, include=TRUE}

##Load data
load(file = "4.- BRCA_CNV_TM_genes_GR.rda") #load genes_GR and genes
load(file = "4.-BRCA_CNV_TM_results.rda") # TM_gaia_results, RecCNV_2, threshold


## Select the required column
sCNV <- RecCNV_2[RecCNV_2[,"q-value"] <= threshold,c(1:4,6)]
sCNV <- sCNV[order(sCNV[,3]),]
sCNV <- sCNV[order(sCNV[,1]),]
colnames(sCNV) <- c("chr","Aberration","start","end","q-value")

sCNV <- as.data.frame(sCNV)

##reordered: "chr", "start", "end", "extra1", "extra2"
sCNV <- sCNV[,c(1,3,4,2,5)]

##Hi ha start - end que end es menor que start. S'eliminen
remove <- which((sCNV$end - sCNV$start) < 0)
sCNV <- sCNV[-remove,]

### Convert to GenomicRanges format
sCNV_GR <- makeGRangesFromDataFrame(sCNV, keep.extra.columns = TRUE)

## Save
save(sCNV_GR,sCNV,file = "4.- BRCA_CNV_TM_sCNV_GR.rda", compress = "xz")



#_____________ Overlap with reference dataset 

## Find overlapping intervals
hits <- findOverlaps(genes_GR, sCNV_GR, type = "within")
sCNV_ann <- cbind(sCNV[subjectHits(hits),], genes[queryHits(hits),])
rownames(sCNV_ann) <- NULL
head(sCNV_ann)

AberrantRegion <- paste0(sCNV_ann[,1],":",sCNV_ann[,3],"-",sCNV_ann[,4])
GeneRegion <- paste0(sCNV_ann[,7],":",sCNV_ann[,8],"-",sCNV_ann[,9])
AmpDel_genes <- cbind(sCNV_ann[,c(6,1,4,5)],AberrantRegion,GeneRegion)

AmpDel_genes[AmpDel_genes$Aberration == "0", "Aberration"] <- "Del"
AmpDel_genes[AmpDel_genes$Aberration == "1", "Aberration"] <- "Amp"

dim(AmpDel_genes)#4288
table(AmpDel_genes$Aberration)
# Amp  Del 
#2733 1555

#There are many repeated genes with a number after a guide. It is removed
AmpDel_genes_dp<- AmpDel_genes
AmpDel_genes_dp$GeneSymbol <- gsub("\\-.*", "\\", AmpDel_genes$GeneSymbol)

#Duplicate delation
AmpDel_genes_dp<- AmpDel_genes_dp[!duplicated(AmpDel_genes_dp$GeneSymbol), ]
dim(AmpDel_genes_dp)#2221

table(AmpDel_genes_dp$Aberration, AmpDel_genes_dp$chr )
#        6   9  13  14  15  16  20  23
#  Amp   2 463   0   0   0 372   0 462
#  Del   0   0 349 540  10   0  23   0

table(AmpDel_genes_dp$Aberration )
# Amp  Del 
#1299  922

table(AmpDel_genes_dp$chr)
#  6   9  13  14  15  16  20  23 
#  2 463 349 540  10 372  23 462


## Save
save(AmpDel_genes,file = "4.- BRCA_CNV_TM_AmpDel_genes.rda")
load(file = "4.- BRCA_CNV_TM_AmpDel_genes.rda")

write.table(AmpDel_genes, file="Plots_results/4.- 4.- BRCA_CNV_TM_AmpDel_genes.csv", sep=",", row.names=F)

## Sense duplicats
save(AmpDel_genes_dp,file = "4.- BRCA_CNV_TM_AmpDel_genes_dp.rda")
load(file = "4.- BRCA_CNV_TM_AmpDel_genes.rda")

write.table(AmpDel_genes_dp, file="Plots_results/4.- 4.- BRCA_CNV_TM_AmpDel_genes_dp.csv", sep=",", row.names=F)


```








